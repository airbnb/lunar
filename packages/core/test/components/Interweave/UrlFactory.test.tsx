import React from 'react';
import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import UrlFactory from '../../../src/components/Interweave/factories/Url';

describe('linkFactory()', () => {
  const urlParts = {
    auth: '',
    fragment: '',
    host: '',
    path: '',
    port: '',
    query: '',
    scheme: '',
  };

  it('renders a link', () => {
    const wrapper = shallowWithStyles(<UrlFactory urlParts={urlParts}>airbnb.com</UrlFactory>);

    expect(wrapper).toMatchSnapshot();
  });

  it('adds http:// to URL', () => {
    const wrapper = shallowWithStyles(
      <UrlFactory urlParts={urlParts}>airbnb.com</UrlFactory>,
      true,
    );

    expect(wrapper.prop('href')).toBe('http://airbnb.com');
  });

  it('doesnt add http:// to URL', () => {
    const wrapper = shallowWithStyles(
      <UrlFactory urlParts={urlParts}>https://airbnb.com</UrlFactory>,
      true,
    );

    expect(wrapper.prop('href')).toBe('https://airbnb.com');
  });

  it('sets open in new window', () => {
    const wrapper = shallowWithStyles(
      <UrlFactory urlParts={urlParts} newWindow>
        airbnb.com
      </UrlFactory>,
      true,
    );

    expect(wrapper.prop('openInNewWindow')).toBe(true);
  });

  it('sets small prop on link', () => {
    const wrapper = shallowWithStyles(
      <UrlFactory urlParts={urlParts} small>
        airbnb.com
      </UrlFactory>,
      true,
    );

    expect(wrapper.prop('small')).toBe(true);
    expect(wrapper.prop('large')).toBe(false);
  });

  it('sets large prop on link', () => {
    const wrapper = shallowWithStyles(
      <UrlFactory urlParts={urlParts} large>
        airbnb.com
      </UrlFactory>,
      true,
    );

    expect(wrapper.prop('large')).toBe(true);
    expect(wrapper.prop('small')).toBe(false);
  });
});
