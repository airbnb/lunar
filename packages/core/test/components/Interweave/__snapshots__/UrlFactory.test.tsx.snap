// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`linkFactory() renders a link 1`] = `
<Link
  baseline={true}
  cx={[Function]}
  href="http://airbnb.com"
  openInNewWindow={false}
  withStylesClasses={
    Array [
      "link",
      "link_block",
      "link_baseline",
      "link_inverted",
      "link_muted",
      "link_disabled",
    ]
  }
>
  airbnb.com
</Link>
`;
